purchase_product_weight/models/purchase.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    weight_total = fields.Float(string='Total weight')
+    product_weight = fields.Float(
+        related='product_id.weight',
+        string='Product Weight')
+
+    @api.onchange('product_id', 'product_weight', 'product_qty')
+    def _compute_total_weight(self):
+        for line in self:
+            line.weight_total = line.product_id.weight * \
+                line.product_qty
+
+    @api.onchange('product_qty', 'product_uom', 'weight_total')
+    def _onchange_quantity(self):
```

Comment:
Overall, I think this part of code is quite complicate, I can't understand it easily.
Can make as sub method call with some description on it for easy reading?

Reply:
I..e, a routing for date_planned, and a routing for price_unit

Reply:
Ok, I removed date_planned, which is already computed in original onchange called with super() (the function is copied-pasted and modified from the original one)